src/data/BaseModel.js

```diff
@@ -0,0 +1,153 @@
+        return field.editable === undefined || field.editable !== false;
+    };
+
+    /**
+     * Checks whether or not this field should be excluded from the form.
+     *
+     * @param {Object} field
+     * @returns {boolean}
+     */
+    exclude = (field) => {
+        return field.excludeFromForm !== undefined && field.excludeFromForm === true;
+    };
+
+    /**
+     * Checks whether or not this field should be hidden on the data table.
+     *
+     * @param {Object} field
+     * @returns {boolean}
+     */
+    hide = (field) => {
+        return field.hide !== undefined && field.hide === true;
+    };
+
+    /**
+     * Retrieves the label from a field object, or dynamically
+     * generates it based on the name if the label field does not exist.
+     *
+     * @param {Object} field
+     * @returns {string}
+     */
+    getLabel = (field) => {
+        if (field.label === undefined) {
+            const header = field.name.charAt(0).toUpperCase() + field.name.slice(1);
+            return header.replace(/_/g, " ");
+        }
+
+        return field.label;
+    };
+
+    /**
+     * Checks whether or not this field be shown based on
+     * the item passed through to the `field.conditional(item)` function.
+     *
+     * @param {Object} field
+     * @param {Object} item
+     * @returns {boolean}
+     */
+    isConditional = (field, item) => {
+        if (field.conditional !== undefined) {
+            return field.conditional(item);
+        } else {
+            return false;
+        }
+    };
+
+    /**
+     * Sets a value in a field instance (found by the name given).
+     *
+     * @param {string} name
+     * @param {any} value
+     */
+    set(name, value) {
+        this.fields.map((field) => {
+            if (field.name === name) {
+                field.value = value;
+            }
+
+            return field;
+        });
+    }
+
+    /**
+     * Get a field object from the fields array.
+     * @param name
+     * @returns {Object}
+     */
+    get(name) {
+        return this.fields.find((field) => field.name === name);
+    }
+
+    /**
+     * Returns a default, single level object for use in a form.
+     * This object is dynamically generated from the field configuration passed through.
+     *
+     * @returns {Object}
+     */
+    getDefaultObject() {
+        let object = {};
+        this.fields.forEach((field) => {
+            if (this.isEditable(field)) {
+                object[field.name] = field.defaultValue ? field.defaultValue : undefined;
+            }
+        });
+        return object;
+    }
+
+    /**
+     * Recursively retrieves a value from a data item.
+     * It can get single level values like `data['name']`,
+     * or multi-level values like `data[profile']['first_name']`.
+     *
+     * @param item
+     * @param name
+     * @returns {any}
+     */
+    getValue = (item, name) => {
+        return name.split('.').reduce((o, i) => {
+            return o[i];
```

Comment:
@mtdalton Do you not think we need to have a check here for _undefined_?

Reply:
We definitely do. I'll add this.